paimon-format/src/main/java/org/apache/paimon/format/avro/AvroBytesStringMap.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.format.avro;
+
+import org.apache.paimon.data.BinaryArray;
+import org.apache.paimon.data.BinaryArrayWriter;
+import org.apache.paimon.data.InternalArray;
+import org.apache.paimon.data.InternalMap;
+import org.apache.paimon.types.DataTypes;
+import org.apache.paimon.utils.IntArrayList;
+
+import org.apache.avro.io.BinaryData;
+import org.apache.avro.io.BinaryDecoder;
+
+import java.io.IOException;
+
+/** An {@link InternalMap} stored in avro format bytes. */
+public class AvroBytesStringMap implements InternalMap {
+
+    private static final int ELEMENT_SIZE =
+            BinaryArray.calculateFixLengthPartSize(DataTypes.STRING());
+
+    // map size
+    private int size;
+    // stores original bytes from avro files
+    private byte[] bytes;
+    // total bytes length
+    private int lengthInBytes;
+    // offset of each string
+    private final IntArrayList off;
+    // length of each string
+    private final IntArrayList len;
+
+    private BinaryArray keyArray;
+    private BinaryArray valueArray;
+
+    public AvroBytesStringMap(BinaryDecoder decoder, boolean valueNullable) throws IOException {
```

Comment:
Can you create a static method for this? And just pass all fields to this constructor.

In this way, `AvroBytesStringMap` can be an immutable class.